x-pack/plugins/beats_management/common/io_ts_types.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+
+export class DateFromStringType extends t.Type<Date, string, t.mixed> {
+  // tslint:disable-next-line
+  public readonly _tag: 'DateFromISOStringType' = 'DateFromISOStringType';
+  constructor() {
+    super(
+      'DateFromString',
+      (u): u is Date => u instanceof Date,
+      (u, c) => {
```

Comment:
Some of the variable names in this function make it a little difficult to reason about. It's not unreadable or anything, but the return line in particular made it kind of hard for me to immediately understand what was happening, because I needed to recall four separate symbols with discrete names. Renaming `d` something like `date` and `s` something like `str` may help to alleviate the problem, especially for people unfamiliar with `io-ts` (like me 😄).

Just my 2 cents.